sw/device/lib/dif/dif_padctrl.h

```diff
@@ -0,0 +1,196 @@
+// Copyright lowRISC contributors.
+// Licensed under the Apache License, Version 2.0, see LICENSE for details.
+// SPDX-License-Identifier: Apache-2.0
+
+#ifndef OPENTITAN_SW_DEVICE_LIB_DIF_PADCTRL_H_
+#define OPENTITAN_SW_DEVICE_LIB_DIF_PADCTRL_H_
+
+#include "sw/device/lib/base/mmio.h"
+
+// TODO - Should be autogenerated.
+// - Should padctrl/pinmux share some defines/constants (such as MIO_NUM) or do
+// we just autogen the same number twice?
+#define DIF_PADCTRL_DIO_NUM 32
+#define DIF_PADCTRL_MIO_NUM 32
```

Comment:
Both the DIO and MIO numbers are wrong, they should be 4 and 16 respectively, attr width is correct

Reply:
Yeah, you are right - these numbers will be different from pinmux. Pinmux allows you to control up to 32 MIOs, but I think as you have pointed out - only 16 are connected to the chip pins in padctrl (don't remember exact details but roughly something like that).

These defines will get auto generated, after my recent gen_cheader change.